src/Nethermind/Nethermind.Core/Caching/LruKeyCache.cs

```diff
@@ -84,28 +79,88 @@ public void Delete(TKey key)
         {
             if (_cacheMap.TryGetValue(key, out LinkedListNode<TKey>? node))
             {
-                _lruList.Remove(node);
+                Remove(node);
                 _cacheMap.Remove(key);
             }
         }
 
+        private void MoveToMostRecent(LinkedListNode<TKey> node)
```

Comment:
These methods looks pretty similar in all the implementations. Have you considered making them static and move to `LinkedListNode`? It could accept a field from a cache as `ref first` and would allow remove a lot of similar code?

Reply:
Done